mesa/visualization/components/matplotlib.py

```diff
@@ -43,48 +85,157 @@ def SpaceMatplotlib(
+        agent_portrayal: A callable that returns a dict specifying how to show the agent
+        propertylayer_portrayal: a dict specifying how to show propertylayer(s)
+        ax: the axes upon which to draw the plot
+        post_process: a callable called with the Axes instance
+        postprocess: a user-specified callable to do post-processing called with the Axes instance. This callable
+        can be used for any further fine-tuning of the plot (e.g., changing ticks, etc.)
+        space_drawing_kwargs: any additional keyword arguments to be passed on to the underlying function for drawing the space.
+
+    Returns:
+        Returns the Axes object with the plot drawn onto it.
+
+    ``agent_portrayal`` is called with an agent and should return a dict. Valid fields in this dict are "color",
+    "size", "marker", and "zorder". Other field are ignored and will result in a user warning.
+
+    """
+    if ax is None:
+        fig, ax = plt.subplots()
+
     # https://stackoverflow.com/questions/67524641/convert-multiple-isinstance-checks-to-structural-pattern-matching
     match space:
-        case mesa.space._Grid():
-            _draw_grid(space, space_ax, agent_portrayal, propertylayer_portrayal, model)
+        case mesa.space._Grid() | OrthogonalMooreGrid() | OrthogonalVonNeumannGrid():
+            draw_orthogonal_grid(space, agent_portrayal, ax=ax, **space_drawing_kwargs)
+        case HexSingleGrid() | HexMultiGrid() | mesa.experimental.cell_space.HexGrid():
+            draw_hex_grid(space, agent_portrayal, ax=ax, **space_drawing_kwargs)
+        case mesa.space.NetworkGrid() | mesa.experimental.cell_space.Network():
+            draw_network(space, agent_portrayal, ax=ax, **space_drawing_kwargs)
         case mesa.space.ContinuousSpace():
-            _draw_continuous_space(space, space_ax, agent_portrayal, model)
-        case mesa.space.NetworkGrid():
-            _draw_network_grid(space, space_ax, agent_portrayal)
+            draw_continuous_space(space, agent_portrayal, ax=ax)
         case VoronoiGrid():
-            _draw_voronoi(space, space_ax, agent_portrayal)
-        case Grid():  # matches OrthogonalMooreGrid, OrthogonalVonNeumannGrid, and Hexgrid
-            # fixme add a separate draw method for hexgrids in the future
-            _draw_discrete_space_grid(space, space_ax, agent_portrayal)
-        case None:
-            if propertylayer_portrayal:
-                draw_property_layers(space_ax, space, propertylayer_portrayal, model)
+            draw_voroinoi_grid(space, agent_portrayal, ax=ax)
 
-    solara.FigureMatplotlib(
-        space_fig, format="png", bbox_inches="tight", dependencies=dependencies
-    )
+    if propertylayer_portrayal:
+        draw_property_layers(space, propertylayer_portrayal, ax=ax)
+
+    if post_process is not None:
+        post_process(ax=ax)
 
+    return ax
 
-def draw_property_layers(ax, space, propertylayer_portrayal, model):
+
+def draw_property_layers(
```

Comment:
Note to self: Make singular `draw_property_layer` function, which returns a single axes. Multiple axes with one (semi-transparent) propertylayer each, and optionally a space, can than be combined into a single figure.

Reply:
Not sure I understand the idea